### PR TITLE
Bug Fixes for 3.3.1

### LINKED ISSUE #1312: Load libraries with SO version on Linux and OSX.
Right now unified backend loads libafXYZ.{so,dylib} instead of libafXYZ.{so.3, 3.dylib}.

### PR DESCRIPTION
**Pull request by pavanky**
- Fixed getting device pointer out of OpenCL backend
- Fixed unified backend to load SO versioned libraries

------------------------------------------
- Fixes #1312

### CODE REVIEW

File: src/api/unified/symbol_manager.cpp [hunk 1]
```diff
@@ -27,9 +27,9 @@ static const string LIB_AF_BKND_SUFFIX = ".dll";
 #else
 static const string LIB_AF_BKND_PREFIX = "libaf";
 #if defined(__APPLE__)
-static const string LIB_AF_BKND_SUFFIX = ".dylib";
+static const string LIB_AF_BKND_SUFFIX = ".3.dylib";
 #else
-static const string LIB_AF_BKND_SUFFIX = ".so";
+static const string LIB_AF_BKND_SUFFIX = ".so.3";
```
Comment (9prady9) [inline src/api/unified/symbol_manager.cpp]: @pavanky  May  be we can append the version number programmatically instead of hard coding it ?

Comment (pavanky) [inline src/api/unified/symbol_manager.cpp]: @9prady9 That makes sense. Let me resend the PR.

**Comment by pavanky** [comment]
@9prady9 updated.

**Comment by 9prady9** [comment]
do you want to build it or shall i go ahead and merge it ?

**Comment by pavanky** [comment]
@9prady9 build on linux and osx before merge.

**Comment by pavanky** [comment]
build arrayfire linux ci

**Comment by pavanky** [comment]
build arrayfire osx ci

**Comment by pavanky** [comment]
build arrayfire linux ci

build arrayfire osx ci

**Comment by shehzan10** [comment]
build arrayfire osx ci

**Pull request merged**